### PR TITLE
Run Acceptance Tests on Travis

### PR DESCRIPTION
**Pull request by muhammad-ammar**
TE-497

### CODE REVIEW

**Comment by muhammad-ammar** [comment]
@clintonb Currently two tests are failing. One test has an error and the second is an actual failure. Any thoughts?

**Comment by clintonb** [comment]
@muhammad-ammar Holy cow, this is amazing!!! (The three exclamation points show you how excited I am!)

Regarding the broken tests:
- test_login will fail if you don't have an LMS server. I doubt we want to go through the trouble of setting up a server, so I would just mark the test for skipping unless and environment variable, say `ENABLE_LMS_TESTS` is set to True.
- The API_SERVER_URL failure is due to test "leakage". Nose is pulling in the course_validation package when it shouldn't. See the `accept` target in the Makefile for an example of how I exclude this directory.

File: runAcceptance.sh [hunk 1]
```diff
@@ -2,19 +2,33 @@
...
+cd -
+
+echo "Starting Analytics Data API Server..."
+./edx-analytics-data-api/manage.py runserver 9001 --noreload &
+
+echo "Starting Analytics Dashboard Server..."
+./analytics_dashboard/manage.py runserver 9000 --noreload &
 
 echo "Running acceptance tests..."
 cd acceptance_tests && nosetests
```
Comment (clintonb) [inline runAcceptance.sh]: Either run `make accept` or update this to exclude the course_validation directory:
`nosetests acceptance_tests --exclude-dir=acceptance_tests/course_validation`

**Comment by muhammad-ammar** [comment]
@clintonb Skipped the LMS test. Now build is passing.
BTW how you test test_login? I mean how you prepare the environment for this test?

**Comment by clintonb** [comment]
1. Setup LMS as normal.
2. Add an OAuth2 client via management command (which I don't think
   exists).
3. Add the OAuth credentials to the dashboard settings.

That should be it.

## 

Clinton Blackburn

**Comment by muhammad-ammar** [comment]
@clintonb Can we merge this PR in its current state?

**Comment by clintonb** [comment]
@muhammad-ammar Go for it.

**Comment by muhammad-ammar** [comment]
@clintonb After rebase i am getting errors. On my system, I set the env variables in [**init**.py](https://github.com/edx/edx-analytics-dashboard/blob/master/acceptance_tests/__init__.py) but after that 4 tests are failing. [Here](https://gist.github.com/muhammad-ammar/e9f8b9805c2ab975187f) is the traceback of failing tests.

I am not sure why i am getting these issue as earlier there are no issues like above?

**Comment by clintonb** [comment]
@muhamad-ammar You need to set the following environment variables in runA
cceptance.sh. ENABLE_AUTO_AUTH to True, ENABLE_AUTH_TESTS to False,
and ENABLE_ERROR_PAGE_TESTS.

On Tuesday, October 14, 2014, Muhammad Ammar notifications@github.com
wrote:

> @clintonb https://github.com/clintonb After rebase i am getting errors.
> On my system, I set the env variables in _init_.py
> https://github.com/edx/edx-analytics-dashboard/blob/master/acceptance_tests/__init__.py
> but after that 4 tests are failing. Here
> https://gist.github.com/muhammad-ammar/e9f8b9805c2ab975187f is the
> traceback of failing tests.
> 
> I am not sure why i am getting these issue as earlier there are no issues
> like above?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/edx/edx-analytics-dashboard/pull/154#issuecomment-59036520
> .

## 

Clinton Blackburn

**Comment by muhammad-ammar** [comment]
@clintonb Tests are fixed now. I will merge the PR

**Comment by benpatterson** [comment]
@muhammad-ammar when you have a moment, could you let the analytics team know that acceptance tests are now configured on Travis? Thanks a lot for getting this done & solving the various hitches!

**Comment by muhammad-ammar** [comment]
@benpatterson Yeah, No Problem.

**Pull request merged**